### PR TITLE
model: manifests can be compared

### PR DESCRIPTION
**Pull request by jazzdan**

### CODE REVIEW

**Comment by jazzdan** [comment]
Should I add another test that fails if the structure of Manifest has changed so that we know to update the equality function?

File: internal/model/manifest.go [hunk 1]
```diff
@@ -102,6 +103,14 @@ func (m Manifest) validate() *ValidateErr {
 	return nil
 }
 
+func (m1 Manifest) Equal(m2 Manifest) bool {
+	primitivesMatch := m1.Name == m2.Name && m1.K8sYaml == m2.K8sYaml && m1.DockerRef == m2.DockerRef && m1.BaseDockerfile == m2.BaseDockerfile && m1.StaticDockerfile == m2.StaticDockerfile && m1.StaticBuildPath == m2.StaticBuildPath
+	cmdMatch := m1.Entrypoint.Equal(m2.Entrypoint)
+	pmMatch := m1.FileFilter == m2.FileFilter
```
Comment (nicks) [inline internal/model/manifest.go]: i don't think this will work for any non-trivial file filter?

Comment (nicks) [reply internal/model/manifest.go]: also, did you forget to compare ConfigFiles?

Comment (nicks) [reply internal/model/manifest.go]: and Mounts

Comment (jazzdan) [reply internal/model/manifest.go]: These are in now.

Comment (jazzdan) [reply internal/model/manifest.go]: (after #493 is landed, this'll have to be changed a bit and then it should be good to merge)

**Pull request merged**